e2e/cypress/integration/api/iam/04_teams_api.spec.ts

```diff
@@ -0,0 +1,329 @@
+                projects: [ '*' ]
+              },
+              {
+                effect: 'ALLOW',
+                actions: [ 'iam:projects:assign' ],
+                projects: [ project1.id ]
+              }
+            ],
+            projects: []
+        }
+      });
+    });
+
+    beforeEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    afterEach(() => {
+      cy.cleanupV2IAMObjectsByIDPrefixes(adminToken, cypressPrefix, ['teams']);
+    });
+
+    describe('POST /apis/iam/v2beta/teams', () => {
+      it('admin can create a new team with no projects', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+      });
+
+      it('admin can create a new team with multiple projects', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [project1.id, project2.id])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(2);
+        });
+      });
+
+      it('admin gets a 404 when it attempts to create ' +
+         'a team with a project that does not exist', () => {
+        cy.request({ ...defaultAdminReq,
+            method: 'POST',
+            failOnStatusCode: false,
+            body: teamWithProjects(teamID, [project1.id, project2.id, 'notfound'])
+        }).then((response) => {
+            expect(response.status).to.equal(404);
+        });
+      });
+
+      it('non-admin with project1 assignment access can create a new team ' +
+      'with no projects', () => {
+        cy.request({ ...defaultNonAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(0);
+        });
+      });
+
+      it('non-admin with project1 assignment access can create ' +
+      'a new team with project1', () => {
+        cy.request({ ...defaultNonAdminReq,
+            method: 'POST',
+            body: teamWithProjects(teamID, [project1.id])
+        }).then((response) => {
+            expect(response.body.team.projects).to.have.length(1);
+        });
+      });
+
+      it('non-admin with project1 assignment access gets a 404 when it attempts to create ' +
+      'a team with a project that does not exist', () => {
+        cy.request({ ...defaultNonAdminReq,
+            method: 'POST',
+            failOnStatusCode: false,
+            body: teamWithProjects(teamID,
+              [project1.id, project2.id, 'notfound'])
+        }).then((response) => {
+            expect(response.status).to.equal(404);
+        });
+      });
+
+      it('non-admin with project1 assignment access cannot create ' +
+      'a new team with other projects (403)', () => {
```

Comment:
`other` projects or `unauthorized` projects? From the test name I can't tell if it's bc the `nonAdmin` doesn't have perms to create teams or doesn't have perms _on those projects_

Reply:
The non-admin token only has `iam:project:assign` on `project1`. I guess this whole thing could be updated to read:

```
requestor with iam:project:assign on project1 only cannot create a team with other project
```

Reply:
ohhhh i see. `requestor with project1 assignment permissions but not project2 permissions cannot create a team with both projects` is what this is saying. Lol it's awkward in all the ways. `requestor with assignment permission on project A but not B cannot create a team with both` . . . `requestor cannot create a team containing one project authorized for assignment and other project not authorized for assignment` 😂